pyfr/plugins/base.py

```diff
@@ -104,54 +104,29 @@ def _invoke_postaction(self, intg, **kwargs):
 
 
 class RegionMixin(object):
-    def _init_writer_for_region(self, intg, nout, prefix, *, fpdtype=None):
-        # Base output directory and file name
-        basedir = self.cfg.getpath(self.cfgsect, 'basedir', '.', abs=True)
-        basename = self.cfg.get(self.cfgsect, 'basename')
-
-        # Data type
-        if fpdtype is None:
-            fpdtype = intg.backend.fpdtype
-        elif fpdtype == 'single':
-            fpdtype = np.float32
-        elif fpdtype == 'double':
-            fpdtype = np.float64
-        else:
-            raise ValueError('Invalid floating point data type')
+    def __init__(self, intg, cfgsect, suffix=None):
+        super().__init__(intg, cfgsect, suffix)
 
         # Region of interest
-        region = self.cfg.get(self.cfgsect, 'region', '*')
+        region = intg.cfg.get(cfgsect, 'region', '*')
 
         # All elements
         if region == '*':
-            mdata = self._prepare_mdata_all(intg, fpdtype, nout, prefix)
-            self._add_region_data = self._add_region_data_all
+            self._prepare_region_data_all(intg)
         # All elements inside a box
         elif '(' in region or '[' in region:
-            box = self.cfg.getliteral(self.cfgsect, 'region')
-            mdata = self._prepare_mdata_box(intg, fpdtype, nout, prefix, *box)
-            self._add_region_data = self._add_region_data_subset
+            box = intg.cfg.getliteral(cfgsect, 'region')
+            self._prepare_region_data_box(intg, *box)
         # All elements on a boundary
         else:
-            mdata = self._prepare_mdata_bcs(intg, fpdtype, nout, prefix,
-                                            region)
-            self._add_region_data = self._add_region_data_subset
-
-        # Construct the file writer
-        return NativeWriter(intg, mdata, basedir, basename)
+            self._prepare_region_data_bcs(intg, region)
```

Comment:
`from pyfr.writers.native import NativeWriter` should be removed I think

Reply:
Fixed.